example/src/components/Well.jsx

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import { COLORS } from '../constants/tetrimino';
+import { getExactPosition } from '../lib/grid';
+import WellGrid from './WellGrid';
+import Tetrimino from './Tetrimino';
+
+require('./Well.less');
+
+class Well extends React.Component {
+  /**
+   * A rectangular vertical shaft, where Tetriminos fall into during a Flatris
+   * game. The Well has configurable size and speed. Tetrimino pieces can be
+   * inserted inside the well and they will fall until they hit the bottom, and
+   * eventually fill it. Whenever the pieces form a straight horizontal line it
+   * will be cleared, emptying up space and allowing more pieces to enter
+   * afterwards.
+   */
+  shouldComponentUpdate(nextProps) {
+    return (
+      nextProps.grid !== this.props.grid ||
```

Comment:
I think this will always be true because there are complex data structures that are compared. I think you should use some type of deep comparison like `_.isEqual()`.
Eg for the grid:
![image](https://cloud.githubusercontent.com/assets/1254386/18709751/4f6003f8-800a-11e6-9075-6134627e4e6a.png)

Am I wrong?